velero-plugins/pod/restore.go

```diff
@@ -29,6 +31,64 @@ func (p *RestorePlugin) AppliesTo() (velero.ResourceSelector, error) {
 	}, nil
 }
 
+// Returns true if pod is not excluded by namespace rules
+func (p *RestorePlugin) podRestoreHookIncludeNamespace(pod *corev1API.Pod, restoreHookSpec velerov1.RestoreResourceHookSpec) bool {
+	if len(restoreHookSpec.IncludedNamespaces) == 0 && !common.StringInSlice(pod.Namespace, restoreHookSpec.ExcludedNamespaces) {
+		p.Log.Info("[pod-restore] pod has restore hooks via unspecified included namespaces and is not in excluded namespaces")
+		return true
+	}
+	if common.StringInSlice(pod.Namespace, restoreHookSpec.IncludedNamespaces) {
+		p.Log.Info("[pod-restore] pod has restore hooks via included namespaces")
+		return true
+	}
+	p.Log.Info("[pod-restore] pod has no restore hooks via included/exclude namespaces")
+	return false
+}
+
+func (p *RestorePlugin) podRestoreHookIncludeResources(pod *corev1API.Pod, restoreHookSpec velerov1.RestoreResourceHookSpec) bool {
+	if len(restoreHookSpec.IncludedResources) == 0 && !common.StringInSlice(pod.Name, restoreHookSpec.ExcludedResources) {
```

Comment:
You're comparing the pod Name field rather than the resource. If you look at the Velero code in item_hook_handler.go, they start with the groupResource for the item and then use velero's util/collections IncludesExcludes.ShouldInclude() func to determine whether the current resource should be included. It might be easier to import that and use it rather than reimplementing, but in any case it needs to be based on the current resource, not the current item's name.

Reply:
+1

Reply:
Wanted to import hook but its internal..  `could not import github.com/vmware-tanzu/velero/internal/hook (invalid use of internal package github.com/vmware-tanzu/velero/internal/hook)compilerBrokenImport`

Reply:
@kaovilai We may need to just copy and paste some stuff, or at least look at their logic and handle the same edge cases here.